app/static/js/script.js

```diff
@@ -228,34 +227,36 @@ function downloadImage(selector, svg_selector, png) {
 	}
 
 	if(graph.attr('id') === 'bubbleGraph') {
-		downloadSvg.attr('height', '550').attr('width', '950');
-		downloadSvg.select('rect.white-rect').attr('height', '550').attr('width', '950');
+		downloadSvg.attr('height', '550').attr('width', width+240); //from 950 to width+240
+		downloadSvg.select('rect.white-rect').attr('height', '550').attr('width', width+240);//from 950 to width+240
+	    heightPos = 470; //add this line
 	}
 
 	if(graph.attr('id') === 'timeSeries') {
-		downloadSvg.attr('height', '550').attr('width', '700');
-		downloadSvg.select('rect.white-rect').attr('height', '550').attr('width', '700');
+		downloadSvg.attr('height', '550').attr('width', width+250); //from 950 to width+250
```

Comment:
I didn't change the heights for each graph since:
1. the browser height is **fixed** for each graph no matter how you change the window size, but the width is not.
2. each height fits well when downloading (no over spilling or huge blank area)

As mentioned, the browser width is not fixed. However, the downloaded picture size dependents on the browser displayed picture size, which makes the fixed-width not work here: if the displayed size is too large, the graph will be incomplete (spillover)  and if the displayed size is small, a huge blank appears. 

With the flexible width, we can have a reasonable width for the citation text while guaranteeing a complex graph in the .svg/.png file. 

Besides, I do think `redraw the whole image from scratch inside the downloadsvg element using the existing drawing functions passing the id of a g graphics element to draw onto, inside a larger canvas with the logo and cite the text.` is a better way to fix the width and height for each graph (or we can consider add the attribution of `scale` here?  as what I've done in `worldMap`) but I'm not very sure how to realize it... if needed, I will explore a bit!

Reply:
This one is a bit open ended, it might be better if I open a new issue and we come back to it later. I need to implement this on some other sites and trying to think about the best way to do it.

My main concern here at the moment is we have a lot of fixed values for sizes and positions, which given the width is not fixed means you can still break the output with certain screen dimensions. Its also harder to change and update the graphs later with fixed rather than calculated values. But that leads me to conclude that it needs to be rewritten somehow so theres not much point spending anymore time fixing the current version.

So I think we close this now and open an issue for changing how it works to either a complete redraw or a transform (the graph we have drawn is vector graphics so it should be possible to resize the whole graph though if the aspect ratio isnt preserved some shapes might distort, which makes the redraw perhaps the best solution).